gson/src/test/java/com/google/gson/FieldNamingPolicyTestCase.java

```diff
@@ -0,0 +1,250 @@
+package com.google.gson;
+
+import com.diffblue.deeptestutils.Reflector;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class FieldNamingPolicyTestCase {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: com/google/gson/FieldNamingPolicy.java */
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - com/google/gson/FieldNamingPolicy.java:148: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 149 branch to line 149
+   *     - conditional line 150 branch to line 150
+   *     - conditional line 150 branch to line 153
+   */
+
+  @Test
+  public void separateCamelCaseInputNotNullNotNullOutputNotNull()
+      throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+
+    // Arrange
+    final String name = "!";
+    final String separator = "$$$$$$$";
```

Comment:
Heuristics could possibly help us: something called 'separator' should be a single character, preferably ',', ';', etc.
Can you create a ticket for this please?

Reply:
Will do.

Reply:
Yes, the point is that we shouldn't only test the corner cases, but also the 'normal' case.